ui/components/app/asset-list-item/asset-list-item.js

```diff
@@ -68,13 +68,17 @@ const AssetListItem = ({
           e.stopPropagation();
           sendTokenEvent();
           dispatch(
-            updateSendToken({
-              address: tokenAddress,
-              decimals: tokenDecimals,
-              symbol: tokenSymbol,
+            updateSendAsset({
```

Comment:
This function now results in a token balance check before resolving, which can be quite slow. Whereas before this function only performed synchronous work, it'll now be async and dependent upon the network, so it might take much longer. I'm concerned that this might give the appearance that the button hasn't worked (particularly on slow networks), or that the user might be more likely to click this multiple times.

Reply:
The same concern applies to the token overview page as well

Reply:
loading indicator will now be displayed when dealing with tokens/right before the balance check. fixed in https://github.com/MetaMask/metamask-extension/pull/10965/commits/f94694c0f55a9ccdfe96a7563f4acb4acb56d70b